src/gsi/read_fed.f90

```diff
@@ -156,223 +146,100 @@ subroutine read_fed(nread,ndata,nodata,infile,obstype,lunout,twind,sis,nobs)
-     status = nf90_inq_varid( ncdfID, 'time', varID )
-     status = nf90_get_var( ncdfID, varID, utime )
-
-     ! CLOSE NETCDF FILE
-     status = nf90_close( ncdfID )
-
-
-     !-Obtain analysis time in minutes since reference date
-     sec70 = 694267200.0  ! seconds since from 1978-01-01 00:00 to 2000-01-01 12:00    
-                          ! because the official GOES prescribed epoch time for GLM data is 2000-01-01 12:00:00
-
-     call w3fs21(iadatemn,mins_an)  !mins_an -integer number of mins snce 01/01/1978
-     rmins_an=mins_an             !convert to real number
-
-     ! SINCE ALL OBS WILL HAVE THE SAME TIME, CHECK TIME HERE:
-     rmins_ob = ( utime(1) + sec70 )/60   !Convert to Minutes from seconds
-     twindm = twind*60.    !Convert to Minutes from hours
-     timeb = rmins_ob-rmins_an
-
-     if(abs(timeb) > abs(twindm)) then
+      !------------------------
+      ! Allocate data arrays
+      !-------------------------
+      ALLOCATE( fed3d_column( 5, maxobs ) )
+      allocate( fed3d_hgt(maxobs) )
+      ALLOCATE( utime( 1 ) )  ! seconds since from 2000-01-01 12:00
+      fed3d_hgt = -999.0_r_kind 
+
+      !------------------------
+      ! Get useful data arrays
+      !-------------------------
+      ! LON
+      status = nf90_inq_varid( ncdfID, 'lon', varID )
+      status = nf90_get_var( ncdfID, varID, fed3d_column(1, :) )
+      ! LAT
+      status = nf90_inq_varid( ncdfID, 'lat', varID )
+      status = nf90_get_var( ncdfID, varID, fed3d_column(2, :) )
+      ! FED value
+      status = nf90_inq_varid( ncdfID, 'value', varID )
+      status = nf90_get_var( ncdfID, varID, fed3d_column(3, :) )
+      ! TIME
+      status = nf90_inq_varid( ncdfID, 'time', varID )
+      status = nf90_get_var( ncdfID, varID, utime )
+
+      ! FED height, optional variable 
+      status = nf90_inq_varid( ncdfID, 'height', varID )
+      if(status==nf90_noerr)&
+      status = nf90_get_var( ncdfID, varID, fed3d_hgt )
+
+      ! CLOSE NETCDF FILE
+      status = nf90_close( ncdfID )
+
+
+      !-Obtain analysis time in minutes since reference date
+      sec70 = 694267200.0  ! seconds since from 1978-01-01 00:00 to 2000-01-01 12:00    
+                           ! because the official GOES prescribed epoch time for GLM data is 2000-01-01 12:00:00
+
+      call w3fs21(iadatemn,mins_an)  !mins_an -integer number of mins snce 01/01/1978
+      rmins_an=mins_an             !convert to real number
+
+      ! SINCE ALL OBS WILL HAVE THE SAME TIME, CHECK TIME HERE:
+      rmins_ob = ( utime(1) + sec70 )/60   !Convert to Minutes from seconds
+      twindm = twind*60.    !Convert to Minutes from hours
+      timeb = rmins_ob-rmins_an
+
+      if(abs(timeb) > abs(twindm)) then
         print*, 'WARNING: ALL FED OBSERVATIONS OUTSIDE ASSIMILATION TIME WINDOW: ', timeb, twindm
-    !    goto 314
-     endif
-     numfed = maxobs
-     do i=1,numfed
+      endif
+
+      !relative time to write out
+      timeb=real(mins_an-iwinbgn,r_kind)
```

Comment:
@hongli-wang should it be "timeb=real(rmins_ob-iwinbgn,r_kind)"? 
And, the above comment could be " !time relative to the beginning of the da time window"

Reply:
Made changes. Please let me know your other comments/suggestions.